portal_blocks_documentation/docs.py

```diff
@@ -81,48 +101,145 @@ def write_official_doc(doc_json_file: pathlib.Path, block_name: str):
     doc = [DISCLAIMER, f"{doc_json['summary']}"]
 
     if doc_json.get("inputs", False):
-        doc.extend([
-                "### Inputs",
-                "\n".join(doc_json["inputs"])
-        ])
+        doc.extend(["### Inputs", "\n".join(doc_json["inputs"])])
 
     if doc_json.get("output", False):
-        doc.extend([
-            "### Output",
-            "\n".join(doc_json["output"])
-        ])
+        doc.extend(["### Output", "\n".join(doc_json["output"])])
 
-    block_image_url = (f"https://raw.githubusercontent.com/battlefield-portal-community/"
-                       f"Image-CDN/main/portal_blocks/{block_name}.png")
+    block_image_url = (
+        f"https://raw.githubusercontent.com/battlefield-portal-community/"
+        f"Image-CDN/main/portal_blocks/{block_name}.png"
+    )
 
     doc.append(f"\n![{block_name}]({block_image_url})")
 
     doc.append(DISCLAIMER)
     doc_file = BUILD_DIR / block_name / "docs" / "official.md"
     doc_file.write_text("\n".join(doc))
 
-def create_menu_entry(block_name: str):
-    sub_lookup = '# - name: generated-block-reference-do-not-remove'
-    menu_entry_name = f'- name: {block_name}'
-    menu_entry_ref = f'  ref: "/portal-builder/rules-editor/block-reference/{block_name}"'
+
+def cleanup_menu_entries(sub_lookup_start: str, sub_lookup_end: str):
     if MENU_FILE.exists():
-        entries = MENU_FILE.read_text().split('\n')
+        startIndex = -1
+        endIndex = -1
+        entries = MENU_FILE.read_text().split("\n")
```

Comment:
as MENU_FILE is edited quite often... going forward we shall use a yaml parser to load and save data